dts/bindings/gpio/nxp,s32-gpio.yaml

```diff
@@ -1,7 +1,23 @@
-# Copyright 2022 NXP
+# Copyright 2022-2023 NXP
 # SPDX-License-Identifier: Apache-2.0
 
-description: NXP S32 GPIO node
+description: |
+  NXP S32 GPIO controller.
+
+  GPIO can route the external input pad interrupts to the SIUL2 external
+  interrupt controller or the WKPU controller, the latter only when supported by
+  the hardware. By default, GPIO interrupts are routed to SIUL2.
+
+  To route external interrupts to the WKPU controller, the GPIO specifier must
+  be supplied with the flag `NXP_S32_GPIO_INT_WKPU`. For example, the following
+  snippet of DT source instructs the GPIO controller to route the interrupt on
+  pin 9 of `gpioa` to the WKPU controller:
+
```

Comment:
this is probably a good spot to expand on why a user would want to do this and I guess make explicit that there's no wake-up functionalities (for now)

Reply:
added details as requested

Reply:
Thanks!